bodo/ir/iceberg_ext.py

```diff
@@ -875,16 +861,35 @@ def get_filters_pyobject(filter_str, vars):  # pragma: no cover
 
 
 try:
-    import bodo_iceberg_connector as bic
+    import pyiceberg.expressions as pie
 except ImportError:
-    bic = None
+    pie = None
+
+
+def literal(val) -> Literal:
+    """
+    Wrapper over PyIceberg's literal function for constructing filters.
+    This is needed to convert other Python literals to Iceberg-compatible ones
+    """
+    from pyiceberg.expressions.literals import DateLiteral, TimestampLiteral
+    from pyiceberg.expressions.literals import literal as inner_literal
+
+    # PyIceberg literal doesn't support all data types
+    # See https://github.com/apache/iceberg-python/issues/1456
+    if isinstance(val, datetime.date):
+        return DateLiteral((val - datetime.date(1970, 1, 1)).days)
+    if isinstance(val, datetime.datetime):
+        return TimestampLiteral((val - datetime.datetime(1970, 1, 1)).microseconds)
+    if isinstance(val, (list, pd.core.arrays.ExtensionArray)):
```

Comment:
Had to add `pd.core.arrays.ExtensionArray`

Reply:
I'm not sure exactly why, just that sometimes Pandas arrays are used instead of lists especially when coming from BodoSQL

Reply:
The list/array difference is fine, but my point is that we may need other nested types (struct/map) here too. A TODO may be enough for now.